### PR TITLE
Clean up the intrinsics #ifdef mess with a common header

### PR DESCRIPTION
**Pull request by Tilka**

### CODE REVIEW

**Comment by Tilka** [comment]
Also tested locally with `ENABLE_PCH=OFF`.

File: Source/Core/Common/CommonFuncs.h [hunk 1]
```diff
@@ -38,20 +38,6 @@ struct ArraySizeImpl : public std::extent<T>
 	#define __GNUC_PREREQ(a, b) 0
 #endif
 
-#if (defined __GNUC__ && !__GNUC_PREREQ(4,9)) && \
```
Comment (delroth) [inline Source/Core/Common/CommonFuncs.h]: What happened to that?

Comment (Sonicadvance1) [inline Source/Core/Common/CommonFuncs.h]: I'm also wonder this.

Comment (Tilka) [inline Source/Core/Common/CommonFuncs.h]: Put it back in, I must have been confused.

**Pull request merged**